src/compas_rhino/artists/artist.py

```diff
@@ -55,7 +59,7 @@ class Artist(object):
 
     __module__ = "compas_rhino.artists"
 
-    def __init__(self, layer=None):
+    def __init__(self, geometry=None, attributes=None, layer=None):
         self._layer = None
```

Comment:
I know it's not your code, but I always find these two lines redundant. Setting `self._layer = None` has no effect because the next line will set it again to whatever value `layer` has.

Reply:
```python
self._layer = None
self.layer = layer
```
in combination with
```python
@property
def layer(self):
    return self._layer
@layer.setter
def layer(self, layer):
    self._layer = layer
```
is there to allow for processing the value assigned to layer before assigning it.
however, if i am not mistaken, the original version of this used the `property` decorators to allow the user to directly assign to the layer attribute but have the attribute stored in the attributes dict.
```python
self.layer = layer
```
```python
@property
def layer(self):
    return self.attributes['layer']
@layer.setter
def layer(self, layer):
    self.attributes['layer'] = layer
```
since it would make sense to make the layer part of the serialised data. the latter would make sense if we still want the user to have access to the attribute directly...

Reply:
> is there to allow for processing the value assigned to layer before assigning it.

but the thing is always assigned, it's the immediate next line (done via the property setter)
Or am I missing something?